core/src/saros/concurrent/jupiter/internal/Jupiter.java

```diff
@@ -82,8 +82,10 @@ public JupiterActivity generateJupiterActivity(Operation op, User source, SPath
     JupiterActivity jupiterActivity = new JupiterActivity(this.vectorTime, op, source, editor);
 
     // add(op, myMsgs) to outgoing;
-    this.ackJupiterActivityList.add(
-        new OperationWrapper(op, this.vectorTime.getLocalOperationCount()));
+    if (!(op instanceof TimestampOperation)) {
```

Comment:
So you pointed me to this line and what is the problem here ? The problem is that discardAcknowledgedOperations is never called. This was never done in this method anyway.

Reply:
Yeah, you are right `generateJupiterActivity` is not the problem (this is just a rather unnecessary small optimization), but instead, I wanted to point you to `receiveJupiterActivity`. But that is actually a non-issue either, sorry, it has been a while since I was working on this PR.